tkg/client/update_credentials.go

```diff
@@ -197,3 +210,118 @@ func (c *TkgClient) updateVSphereCredentialsForCluster(clusterClient clusterclie
+		return errors.New("either tenantId, clientId or clientSecret should not be empty")
+	}
+
+	if !options.IsRegionalCluster {
+		unifiedIdentity, err := clusterClient.CheckUnifiedAzureClusterIdentity(options.ClusterName, options.Namespace)
+		if err != nil {
+			return err
+		}
+		if unifiedIdentity {
+			log.Warningf("AzureCluster %s use the same AzureClusterIdentity from its management cluster. It cannot be updated separately.", options.ClusterName)
+			return nil
+		}
+	}
+
+	if err := c.UpdateAzureCredentialsForCluster(clusterClient, options, false); err != nil {
+		return err
+	}
+
+	if options.IsRegionalCluster {
+		clusters, err := clusterClient.ListClusters("")
+		if err != nil {
+			return errors.Wrapf(err, "unable to list workload clusters")
+		}
+
+		for i := range clusters {
+			if clusters[i].Name == options.ClusterName {
+				continue
+			}
+			unifiedIdentity, err := clusterClient.CheckUnifiedAzureClusterIdentity(clusters[i].Name, clusters[i].Namespace)
+			if err != nil {
+				log.Error(err, "unable to update credentials for workload cluster")
+				continue
+			}
+			if options.IsCascading || unifiedIdentity {
+				log.V(4).Infof("Updating credentials for workload cluster %q ...", clusters[i].Name)
+				err := c.UpdateAzureCredentialsForCluster(clusterClient, &UpdateCredentialsOptions{
+					ClusterName:               clusters[i].Name,
+					Namespace:                 clusters[i].Namespace,
+					IsRegionalCluster:         false,
+					AzureUpdateClusterOptions: options.AzureUpdateClusterOptions,
+				}, unifiedIdentity)
+				if err != nil {
+					log.Error(err, "unable to update credentials for workload cluster")
+					continue
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *TkgClient) UpdateAzureCredentialsForCluster(clusterClient clusterclient.Client, options *UpdateCredentialsOptions, unifiedIdentity bool) error {
+	if options.IsRegionalCluster {
+		// update capz-manager-bootstrap-credentials
+		log.Infof("Updating secret capz-manager-bootstrap-credentials for management cluster %q", options.ClusterName)
+		if err := clusterClient.UpdateCapzManagerBootstrapCredentialsSecret(options.AzureUpdateClusterOptions.AzureTenantID, options.AzureUpdateClusterOptions.AzureClientID, options.AzureUpdateClusterOptions.AzureClientSecret); err != nil {
+			return err
+		}
+
+		// restart capz-controller-manager pod
+		log.Infof("Restart capz-controller-manager pod")
+		if err := c.RestartCAPZControllerManagerPod(clusterClient); err != nil {
+			return err
+		}
+
+		// UpdateAzureClusterIdentity
+		log.Infof("Update cluster %q AzureClusterIdentity", options.ClusterName)
+		if err := clusterClient.UpdateAzureClusterIdentity(options.ClusterName, options.Namespace, options.AzureUpdateClusterOptions.AzureTenantID, options.AzureUpdateClusterOptions.AzureClientID, options.AzureUpdateClusterOptions.AzureClientSecret); err != nil {
+			return err
+		}
+	} else {
+		if !unifiedIdentity {
```

Comment:
After removing this parameter, please remove the duplicated codes of line 290-293.